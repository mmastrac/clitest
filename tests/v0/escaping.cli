#!/usr/bin/env clitest --v0

$ echo "[LOG] Hello, world!"
! [LOG] %{GREEDYDATA}

$ echo "[LOG] Hello, world!"
! [LOG] Hello, world!

$ echo "[LOG] Hello, world!"
? \[LOG\] %{GREEDYDATA}

$ echo "[LOG] Hello, world!"
? \[%{log=(LOG)}\] %{GREEDYDATA}

$ echo "[LOG] Hello, world!"
%EXPECT_FAILURE
! [LOG]

$ echo "[LOG] Hello, world!"
%EXPECT_FAILURE
! [%{DATA}]

$ echo "[LOG] Hello, world!"
%EXPECT_FAILURE
? \[LOG\]

$ echo "[LOG] Hello, world!"
%EXPECT_FAILURE
? \[%{DATA}\]
