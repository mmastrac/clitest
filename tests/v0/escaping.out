Running <root>/escaping.cli ...

printf "[LOG] Hello, world!\n"
---
[LOG] Hello, world!
---
✅ OK

printf "[LOG] Hello, world!\n"
---
[LOG] Hello, world!
---
✅ OK

printf "[LOG] Hello, world!\n"
---
[LOG] Hello, world!
---
✅ OK

printf "[LOG] Hello, world!\n"
---
[LOG] Hello, world!
---
✅ OK

printf "[LOG] Hello, world!\n"
---
[LOG] Hello, world!
---
✅ OK

printf "[LOG] Hello, world!\n"
---
[LOG] Hello, world!
---
✅ OK

printf "[LOG] Hello, world!\n"
---
[LOG] Hello, world!
---
✅ OK

printf "[LOG] Hello, world!\n"
---
[LOG] Hello, world!
---
✅ OK

<root>/escaping.cli PASSED
1 test(s) passed
